JamesIves/github-pages-deploy-action
# Version 3

**Description**
> Provide a description of what your changes do.

🔴 **Version 3** 🔴 

Migrates the action to use JavaScript instead of bash. There's a few breaking changes here.

- The action will no longer create a `CNAME` file for you automatically.
- The action will no longer run a build script for you.
- You must use the `with` key instead of `env` in your workflow.

With that comes some improvements:

- The action should now deploy much faster.
- It will keep track of your commit history.
- It now utilizes `with` instead of `env`.
- The action can now be properly unit tested. 

I'm currently in the process of finishing this up (mostly just writing tests), but if you'd like to get an early look at this you can point your `workflow.yml` file to `JamesIves/github-pages-deploy-action@beta4` to test it out. 

**Testing Instructions**
> Give us step by step instructions on how to test your changes.

I'll write a proper migration guide when I do the release but here's a quick run down.

* Add a deployment step in your `workflow.yml` file that builds your project. This should occur after checkout, and before this action runs.
* Change `env` to `with`.
* Once the deployment is complete, manually commit a `CNAME` file if you're using a custom domain. You can also safely commit a `.nojekyll` file if need be too.

**Additional Notes**
> Anything else that will help us test the pull request.

Please feel free to flag any issues in this PR.

Closes issue #36 #37 #38
